components/jotform/jotform.app.js

```diff
@@ -0,0 +1,95 @@
+const axios = require('axios')
+const querystring = require('querystring')
+ 
+const jotform = { 
+  type: "app", 
+  app: "jotform", 
+  propDefinitions: {  
+    formId: {
+      // after should be array + assume after apps
+      type: "string", 
+      label: "Form", 
+      // options needs to support standardized opts for pagination
+      async options(opts) { 
+        const forms = await this.getForms(this.$auth.api_key)  
+        // XXX short hand where value and label are same value
+        return forms.content.map(form => { 
+          return { label: form.title, value: form.id } 
+        }) 
+      },
+      // XXX validate  
+    }, 
+  }, 
+  methods: {
+    async _makeRequest(config) {
+      if (config.params) { 
+        const query = querystring.stringify(config.params)
+        delete config.params
+        const sep = config.url.indexOf('?') === -1 ? '?' : '&'
+        config.url += `${sep}${query}`
+        config.url = config.url.replace('?&','?')
+      }
+      try {
+        return await axios(config)
+      } catch (err) {
+        console.log(err) // TODO
+      }
+    },
+    async getForms() {   
+      return (await this._makeRequest({
+        url: `https://api.jotform.com/user/forms`,
+        method: `GET`,
+        headers: {
+          "APIKEY": this.$auth.api_key,
+        },
+      })).data
+    }, 
+    async getWebhooks(opts = {}) {
+      const { formId } = opts
+      return (await this._makeRequest({
+        url: `https://api.jotform.com/form/${formId}/webhooks`,
+        method: `GET`,
+        headers: {
+            "APIKEY": this.$auth.api_key,
+          },
+      })).data
+    },
+    async createHook(opts = {}) {
+      const { formId, endpoint } = opts
+      const updatedEndpoint = `${endpoint}/`
```

Comment:
hopefully doesn't matter if someone passes endpoint with end slash already on there

Reply:
i'll try to check for that....